crates/types/src/item.rs

```diff
@@ -168,18 +125,84 @@ impl TryFrom<&AttributeMap> for Status {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum ItemError {
+    #[error("Invalid Item format")]
+    FormatError,
+    #[error(transparent)]
+    NumParseError(#[from] std::num::ParseIntError),
+    #[error(transparent)]
+    DataParseError(#[from] chrono::format::ParseError),
+}
+
+pub struct Item {
+    pub id: Uuid,
+    pub status: Status,
+    pub created_at: DateTime<Utc>,
+    // TODO: type: Compiling/Verifying
+}
+
+impl Item {
+    pub const fn status_attribute_name() -> &'static str {
+        Status::attribute_name()
+    }
+
+    pub const fn data_attribute_name() -> &'static str {
+        "Data"
+    }
+
+    pub const fn id_attribute_name() -> &'static str {
+        "ID"
+    }
+
+    pub const fn created_at_attribute_name() -> &'static str {
+        "CreatedAt"
+    }
+
+    pub const fn primary_key_name() -> &'static str {
+        Self::id_attribute_name()
+    }
+}
+
+impl From<Item> for AttributeMap {
+    fn from(value: Item) -> Self {
+        let mut item_map = HashMap::from([
+            (
+                Item::id_attribute_name().into(),
+                AttributeValue::S(value.id.into()),
+            ),
+            (
+                Item::created_at_attribute_name().into(),
+                AttributeValue::S(value.created_at.to_rfc3339()),
+            ),
+        ]);
+        item_map.extend(HashMap::from(value.status));
+
+        item_map
+    }
+}
+
 impl TryFrom<AttributeMap> for Item {
     type Error = ItemError;
     fn try_from(value: AttributeMap) -> Result<Item, Self::Error> {
         let id = value
             .get(Item::id_attribute_name())
-            .ok_or(ItemError::FormatError)?;
-        let id = id.as_s().map_err(|_| ItemError::FormatError)?;
+            .ok_or(ItemError::FormatError)?
```

Comment:
I would suggest making more explicit errors here: "ItemError::FormatError("Failed to get id attribute name")

Reply:
Makes sense

Reply:
Done